sceptre/plan/plan.py

```diff
@@ -55,6 +56,25 @@ def _execute(self, *args):
         executor = SceptrePlanExecutor(self.command, self.launch_order)
         return executor.execute(*args)
 
+    def _generate_no_launch_order_error(self):
+        MAX_VALID_STACK_PATH_COUNT = 10
+
+        error_text = f"No stacks detected from the given path '{sceptreise_path(self.context.command_path)}'."
+        command_path = pathlib.PurePath(self.context.command_path)
+        filtered_valid_stack_paths = []
+        for stack_path in self._valid_stack_paths():
+            for parent in pathlib.PurePath(stack_path).parents:
+                if command_path.parent == parent:
+                    filtered_valid_stack_paths.append(stack_path)
+                    break
+            if filtered_valid_stack_paths and len(filtered_valid_stack_paths) > MAX_VALID_STACK_PATH_COUNT:
+                filtered_valid_stack_paths = None
+                break
```

Comment:
A few thoughts on this:
* This seems rather deeply nested. I'd like to avoid getting so nested, especially in this function. It hurts readability. I also don't think we need to do it this way.
* Rather than crawling every combination of segments and comparing parents, I don't think we need to get that intense about it. Rather, I think a simpler and more straight-forward logic here would be to do this:
    1. Check if there's a "/" in the command path. If not, then the prefix to use is `''` (all strings technically start with `''`). If there IS, do `prefix, _ = command_path.rsplit('/' 1)` to get the prefix. 
    2. Filter the valid paths for those that start with the prefix. You can either use `filter()` or a list comprehension. 

Done this way, it would look more like:

```suggestion
        if '/' in self.context.command_path:
            prefix, _ = self.context.command_path.rsplit('/', 1)
        else:
            prefix = ''
        all_paths = self._valid_stack_paths() 
        # We'll fall back to all_paths if no valid path starts with the prefix
        filtered_valid_stack_paths = [p for p in all_paths if p.startswith(prefix + '/')] or all_paths
        if len(filtered_valid_stack_paths) > MAX_VALID_STACK_PATH_COUNT:
            filtered_valid_stack_paths = []
```

Reply:
Refactored function to use this suggestion with `pathlib` rather than `/` processing.